### PR TITLE
Makes door leak damage to components when damaged

### PR DESCRIPTION
**Pull request by comma**
## Description of changes
Previously doors would soak all damage up to its health and only then start passing it to components. This made whole thing kinda meaningless because by the time you start damaging copmonents the door is gone, you can just crowbar it open

Now if door is at 75% health or if it's 10 damage in single hit, some damage can leak through. It's scaled to how broken door is, so you're not going to be crunching locks with first few lucky hits. Secure airlocks override that to keep older behaviour because that's more secure

## Why and what will this PR improve
I think you should be able to bash access locks out of doors without completely destroying the airlock
It opens up access to the components hatch which is kinda pointless if machine is dead

## Authorship
Me

### CODE REVIEW

File: code/game/machinery/doors/_door.dm [hunk 1]
```diff
@@ -325,20 +325,36 @@
...
+	else if(health < maxhealth * 3/4 && initialhealth >= maxhealth * 3/4)
 		visible_message(SPAN_WARNING("\The [src] shows signs of damage!"))
-	..(max(0, damage - initialhealth), damtype)
+
+	..(component_damage, damtype)
 	update_icon()
 
+//How much damage should be passed to components inside even when door health is non zero
+/obj/machinery/door/proc/get_damage_leakthrough(var/damage, damtype=BRUTE)
+	if(health > 3/4 * maxhealth && damage < 10)
```
Comment (MistakeNot4892) [inline code/game/machinery/doors/_door.dm]: `health > (maxhealth * 0.75)` would probably be easier to read.

File: code/game/machinery/doors/_door.dm [hunk 2]
```diff
@@ -325,20 +325,36 @@
...
 /obj/machinery/door/take_damage(var/damage, damtype=BRUTE)
-	var/initialhealth = src.health
-	src.health = max(0, src.health - damage)
-	if(src.health <= 0 && initialhealth > 0)
+	if(!health)
+		..(damage, damtype)
+		update_icon()
+		return
+
+	var/component_damage = get_damage_leakthrough(damage, BRUTE)
```
Comment (MistakeNot4892) [inline code/game/machinery/doors/_door.dm:333]: Should probably return here if `component_damage` is 0.

Comment (comma) [reply code/game/machinery/doors/_door.dm:333]: Nah it's damaging the door health itself if component damage is 0

File: code/game/machinery/doors/_door.dm (hunk 1 above)
Comment (comma) [reply code/game/machinery/doors/_door.dm]: Done

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**